src/lib/routes/mav-stations/index.js

```diff
@@ -0,0 +1,40 @@
+import * as helpers from '../helpers.js';
+
+/**
+ * get train stations from mav-stations API
+ *
+ * @param api mav-station API
+ * @returns list of matching stations
+ */
+const createMavStationsRoute = (api) => async (req, res, next) => {
+  const params = req.query;
+  // default result limit is 10
+  const limit = params.limit ? Number.parseInt(params.limit) : 10;
+  const searchterm = params.query ? helpers.cleanStr(params.query) : undefined;
+  const excludeCountryIso = params.excludeCountryIso
+    ? helpers.cleanStr(params.excludeCountryIso)
+    : undefined;
+
+  await api.stationList().then(function (result) {
+    // filter search term
+    if (searchterm && searchterm.length) {
+      result = result.filter((station) =>
+        helpers.cleanStr(station.name).includes(searchterm)
+      );
+    }
+
+    // filter country
+    if (excludeCountryIso && excludeCountryIso.length) {
```

Comment:
Ich verstehe noch nicht ganz, wie dieser Parameter funktioniert. Standardmäßig werden alle Stationen in Ungarn gefiltert?

Reply:
Ja, wenn es einen Wert bei `excludeCountryIso` gibt, wird dieser genutzt, um ein Land herauszufiltern - in dem Fall Ungarn.

Reply:
Passt.

Reply:
Ich habe es gerade nochmal getestet und verstehe es weiterhin nicht so ganz. Wenn ich beim Startbahnhof einen Bahnhof in Ungarn eintragen will, wird er mir nicht vorgeschlagen, weil er vermutlich herausgefiltert wird. Das ist doch nicht Sinn der Sache, oder?

Reply:
Ja, das ist derzeit so. Vielleicht habe ich die Anforderung nicht richtig verstanden. Meine Annahme: Es sollen alle ungarischen Bahnhöfe herausgefiltert werden, weil das Tool nur in Deutschland genutzt wird. Für Start- und Endbahnhof in Ungarn wird die Selectbox (Trick) genutzt.

Wenn das nicht gewünscht ist, kann ich das gern zurückbauen - oder aus dem Frontend da einfach keinen Wert mitgeben. Wie du magst.

Reply:
Alles klar, jetzt verstehe ich die Verwirrung! 😉

Meine Idee ist folgende: Die meisten Leute würden die Suche vermutlich verwenden, um Verbindungen zu suchen, die nicht in Ungarn starten oder enden. Das ist für eine Suchanfrage aber zwingend notwendig. Daher habe ich den "Trick" eingebaut.

Es sollte darüber hinaus aber möglich sein, auch Verbindungen von und/oder nach Ungarn zu suchen. Dafür gibt es die Option, den Trick zu deaktivieren. Dann müssen aber auch ungarische Stationen im Autocomplete auftauchen.

Mein Vorschlag: Filtern von Stationen im Autocomplete deaktivieren oder rückbauen. Man könnte es zwar vom Trick-Dropdown abhängig machen, in welchem Feld welche Stationen gefiltert werden, aber das fände ich zu verwirrend. Eher würde ich es anders herum machen, dass basierend auf den eingegebenen Stationen Einträge des Tricks entfallen/ausgegraut werden.

Reply:
Okay. Ich hab den Country-Exclude im Autocomplete ausgeschaltet.